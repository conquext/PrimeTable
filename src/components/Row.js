import React from "react";

const Row = ({ rList, className }) => {
  return (
    <div className={className}>
      {rList.map(rl => (
        <button className="nrow" key={rl}>
          {rl}
        </button>
      ))}
    </div>
  );
};

export default Row;
