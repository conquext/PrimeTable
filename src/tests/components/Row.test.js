import React from "React";
import ReactShallowRenderer from "react-test-renderer/shallow";

import Row from "../../components/Row";
import { render } from "@testing-library/react";

const rList = [2, 3, 5, 7, 11];

test("should render Rows correctly", () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Row rList={rList} />);
  //   renderer.render({ component });
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
