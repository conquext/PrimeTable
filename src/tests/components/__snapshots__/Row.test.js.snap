// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Rows correctly 1`] = `
<div>
  <button
    className="nrow"
  >
    2
  </button>
  <button
    className="nrow"
  >
    3
  </button>
  <button
    className="nrow"
  >
    5
  </button>
  <button
    className="nrow"
  >
    7
  </button>
  <button
    className="nrow"
  >
    11
  </button>
</div>
`;
